refactor filters and conditional rendering

connectbo opened the pull request.

trel commented on src/contexts/ServerContext.js:
```diff
@@ -91,36 +91,111 @@ export const ServerProvider = ({ children }) => {
+        }
+        else {
+            let filteredResults = {};
+            let resc_name_query = `SELECT RESC_NAME,RESC_TYPE_NAME,RESC_ZONE_NAME,RESC_VAULT_PATH,RESC_LOC,RESC_INFO, RESC_FREE_SPACE, RESC_COMMENT,RESC_STATUS,RESC_CONTEXT,RESC_PARENT,RESC_ID WHERE RESC_NAME != 'bundleResc' AND RESC_NAME LIKE '%${name}%'`
```
can we use base_query here and just add to it?   same a few lines lower... keep a single source of truth for this set of columns...

trel commented on src/views/resources/ResourceListView.js:
```diff
@@ -68,6 +68,7 @@ export const ResourceListView = () => {
         loadResource((currPage - 1) * perPage, perPage, filterRescName, order, orderBy);
     }, [loadResource, currPage, perPage, filterRescName, order, orderBy])
 
+
```
can remove this once we're happy with the functionality...

trel commented on src/views/resources/ResourceListView.js:
```diff
@@ -146,6 +147,7 @@ export const ResourceListView = () => {
         <Fragment>
             {isLoadingRescContext ? <LinearProgress /> : <div style={{ height: '4px'}}/>}
             <br />
+            {console.log(rescContext)}
```
can remove this once we're happy with the functionality...

trel commented on src/contexts/ServerContext.js:
```diff
@@ -91,36 +91,111 @@ export const ServerProvider = ({ children }) => {
         });
     }
 
+    // const sortResource = (offset, limit, order, orderBy) => {
+    //     resourceSortRemoveDuplicatesHelper(rescContext._embedded, offset, limit, order, orderBy)
+    // }
```
unneeded?

trel reviewed (commented):
looks like it works to me.

only thing is that when filter and there are zero results... might be nice to have a little default thing that says 'No results found'.   Definitely not a blocker.

connectbo commented:
Sorry just saw your comments, too many meetings today :) 

Ok, sounds like the functionality is working, will do some cleaning. I can also create a separate issue for displaying 'No results found', feels like other pages (users, groups) should have this too?

trel commented:
yes, a new issue for 'no results'... thanks.

connectbo commented:
Alright. This pr is ready for review:
- filtering should be working as expected
- added 'no results for [text]'
- optimized the tableview rendering logic

trel commented:
it works 

the 'No results found for []' could be in a 'table' row (white background) - i didn't expect it to be 'lost' in the background grey.

but it's not a blocker if you feel strongly.

connectbo commented:
Alright, I refactored how 'No results found' renders on each page (normal tableview & editing user/group), also changed the textfield placeholder on resource page to 'Filter by name or hostname'. Ready for your review.

![image](https://user-images.githubusercontent.com/33065086/133650266-42dafb4a-8f5b-4325-a260-c2d5b9f6b6ee.png)
![image](https://user-images.githubusercontent.com/33065086/133650333-baad9e8b-f6a5-4ac8-9e89-ab9732a74985.png)

trel commented:
i like it a lot.

I see two tiny inconsistencies...   The column name and the word in the 'Filter' box...

<img width="447" alt="Screen Shot 2021-09-16 at 1 49 26 PM" src="https://user-images.githubusercontent.com/55238/133660592-72cf3a59-9148-4196-a57c-d8425e706d0a.png">

<img width="416" alt="Screen Shot 2021-09-16 at 1 49 32 PM" src="https://user-images.githubusercontent.com/55238/133660609-723d5b3f-6367-4dbb-9693-1494a5a4c071.png">

I propose... 'User Name' and 'Group Name' as the 'correct' way to display these.   We'll leave Hostname out of it for now :)

connectbo commented:
Ahh yes, right. These should be consistent, will change both.

trel commented:
we're really close now...

please make the new field say 'Group Name' too.
<img width="240" alt="Screen Shot 2021-09-16 at 9 06 47 PM" src="https://user-images.githubusercontent.com/55238/133707652-68d340bb-7b60-4afa-9086-2ebf4cb23a8f.png">

and this one say 'User Name'
<img width="218" alt="Screen Shot 2021-09-16 at 9 08 09 PM" src="https://user-images.githubusercontent.com/55238/133707733-8c229260-7b10-471a-bb79-03f510e3c73e.png">

promise that's it!

connectbo commented:
Ohh right, I tried to look for 'Username' and 'GroupName' in my code and found nothing, should have used case-insensitive search. I will get this correct.

trel commented:
confirmed - please # both!

connectbo commented:
yep, added both!
